packages/connection-storage/src/connection-storage.spec.ts

```diff
@@ -230,6 +232,55 @@ describe('ConnectionStorage', function () {
       );
       expect(expectedConnectionInfo).to.deep.equal(connectionInfo);
     });
+
+    context(`max allowed connections ${maxAllowedConnections}`, function () {
+      const createNumberOfConnections = (num: number) => {
+        const connectionInfos = Array.from({ length: num }, (v, i) =>
+          getConnectionInfo({
+            lastUsed: new Date(1690876213077 - (i + 1) * 1000), // Difference of 1 sec
+          })
+        );
+        connectionInfos.forEach((connectionInfo) =>
+          writeFakeConnection(tmpDir, { connectionInfo })
+        );
+
+        return connectionInfos;
+      };
+
+      it('truncates recents to max allowed connections', async function () {
+        const connectionInfos = createNumberOfConnections(
+          maxAllowedConnections
+        );
+
+        const deleteSpy = Sinon.spy(connectionStorage, 'delete');
+
+        // Save another connection
+        await connectionStorage.save(getConnectionInfo());
+
+        const numConnections = (await connectionStorage.loadAll()).length;
+        expect(numConnections).to.equal(maxAllowedConnections);
+
+        expect(
+          deleteSpy.calledOnceWithExactly(
+            connectionInfos[connectionInfos.length - 1].id
+          )
+        ).to.be.true;
+      });
+
+      it('does not remove recent if recent connections are less then max allowed connections', async function () {
+        createNumberOfConnections(8);
```

Comment:
I'm being picky here, but I would use maxAllowedConnections - 1, instead of 8, just in case we change the variable we don't break the test.

Unless we want the test to fail if the maximum number of connections allowed is less than 8.

```suggestion
        createNumberOfConnections(maxAllowedConnections - 1);
```

Reply:
Isn't that the point of tests - any change in code should break it? Let's say if we change `maxAllowedConnections` to 5, it should break the test and highlight that I should update tests as well.

Reply:
I do agree, I don't have a strong opinion on this specific scenario. I was being picky as it will fail under some changes, not all of them. For example, if you change maxAllowedConnections to something greater than 10, the test will still work. It will only fail if it's lower than 8.

But I don't think this is going to change much so I would keep it like that. The test is awesome, fast and gives good feedback.

Reply:
well that's a fair point. let me change it :)

Reply:
added here [`a682345` (#4690)](https://github.com/mongodb-js/compass/pull/4690/commits/a68234595840fb4e7bfc1452027b4dcc3953e1b8)